mime-types.lisp

```diff
@@ -101,3 +103,63 @@ First uses MIME-LOOKUP, then MIME-PROBE and lastly returns the DEFAULT if both f
   "Returns a matching file-extension for the given mime-type.
 If the given mime-type cannot be found, NIL is returned."
   (gethash mime-type *reverse-mime-db*))
+
+(defun mime-equal (m1 m2)
+  "Checks whether M1 and M2 are matching.
+
+In particular, checks the match of type and subtype (any of which can
+be asterisks), discarding any parameters there might be.
+
+\(mime-equal \"text/*\" \"text/html\")
+T
+
+\(mime-equal \"text/html\" \"text/html;charset=utf8\")
+T
+
+\(mime-equal \"*/*\" \"application/octet-stream\")
+T
+
+\(mime-equal \"text/*\" \"application/octet-stream\")
+NIL"
+  (or (equal "*" m1)
+      (equal "*" m2)
+      (equal "*/*" m1)
+      (equal "*/*" m2)
+      (destructuring-bind (type1 subtype1 &rest parameters1)
+          (uiop:split-string m1 :separator '(#\/ #\;))
```

Comment:
Would be better to have this stuff not cons, but I don't particularly care, so it's fine.